res/controllers/Reloop-Digital-Jockey-2-IE.scripts.js

```diff
@@ -0,0 +1,828 @@
+    this.number = number;
+    this.searchMidiCtrl = searchMidiCtrl;
+    this.scratchMidiCtrl = scratchMidiCtrl;
+    this.fxDryWetMidiCtrl = fxDryWetMidiCtrl;
+    this.jogMode = RDJ2.JOGMODES.normal;
+    this.lastNonTraxJogMode = this.jogMode;
+    this.input = this.inputNormal;
+
+    components.Component.call(this);
+    this.updateControls();
+};
+RDJ2.JogModeSelector.prototype = new components.Component({
+    updateControls: function() {
+        var searchValue = this.jogMode === RDJ2.JOGMODES.search ? RDJ2.MIDI_ON : RDJ2.MIDI_OFF;
+        var scratchValue = this.jogMode === RDJ2.JOGMODES.vinyl ? RDJ2.MIDI_ON : RDJ2.MIDI_OFF;
+        var fxDryWetValue = this.jogMode === RDJ2.JOGMODES.fxdrywet ? RDJ2.MIDI_ON : RDJ2.MIDI_OFF;
+
+        if (midi.sendShortMsg) {
+            midi.sendShortMsg(0x90, this.searchMidiCtrl, searchValue);
+            midi.sendShortMsg(0x90, this.scratchMidiCtrl, scratchValue);
+            midi.sendShortMsg(0x90, this.fxDryWetMidiCtrl, fxDryWetValue);
+        }
+    },
+    inputNormal: function(channel, control, value) {
+        var isButtonPressed = RDJ2.isButtonPressed(value);
+        if (isButtonPressed) {
+            switch (control) {
+            case this.searchMidiCtrl:
+                this.jogMode = this.jogMode === RDJ2.JOGMODES.search ? RDJ2.JOGMODES.normal : RDJ2.JOGMODES.search;
+                break;
+            case this.scratchMidiCtrl:
+                this.jogMode = this.jogMode === RDJ2.JOGMODES.vinyl ? RDJ2.JOGMODES.normal : RDJ2.JOGMODES.vinyl;
+                break;
+            case this.fxDryWetMidiCtrl:
+                this.jogMode = this.jogMode === RDJ2.JOGMODES.fxdrywet ? RDJ2.JOGMODES.normal : RDJ2.JOGMODES.fxdrywet;
+                break;
+            default:
+                RDJ2.logError("Unexpected MIDI ctrl value: " + control);
+            }
+            if (this.jogMode !== RDJ2.JOGMODES.vinyl && engine.isScratching(this.number)) {
+                engine.scratchDisable(this.number, RDJ2.JOG_SCRATCH_RAMP);
+            }
+            this.updateControls();
+        }
+    },
+    inputTrax: function(channel, control, value) {
+        var isButtonPressed = RDJ2.isButtonPressed(value);
+        if (isButtonPressed) {
+            switch (control) {
+            case this.searchMidiCtrl:
+            case this.searchMidiCtrl + RDJ2.SHIFT_OFFSET:
+                engine.setValue("[Library]", "MoveRight", 1);
+                break;
+            case this.scratchMidiCtrl:
+            case this.scratchMidiCtrl + RDJ2.SHIFT_OFFSET:
+                engine.setValue("[Library]", "MoveLeft", 1);
+                break;
+            case this.fxDryWetMidiCtrl:
+            case this.fxDryWetMidiCtrl + RDJ2.SHIFT_OFFSET:
+                //'MoveFocusForward' is equivalent to pressing TAB key on the keyboard
+                engine.setValue("[Library]", "MoveFocusForward", 1);
+                break;
+                /* The below code will probably be removed.
```

Comment:
Should this be removed then? Note you might want to consider [controller settings](https://github.com/mixxxdj/mixxx/wiki/midi%20scripting#settings-api) if you wanted to keep that optional.